packages/@aws-cdk/aws-appsync/lib/data-source.ts

```diff
@@ -81,13 +103,41 @@ export interface ExtendedDataSourceProps {
 }
 
 /**
- * Abstract AppSync datasource implementation. Do not use directly but use subclasses for concrete datasources
+ * An Interface for Data Sources
  */
-export abstract class BaseDataSource extends Construct {
+export interface IDataSource extends IConstruct {
+  /**
+   * the name of the data source
+   */
+  readonly name: string;
+  /**
+   * the arn of the data source
+   */
+  readonly arn: string;
+  /**
+   * the underlying CFN data source resource
+   */
+  readonly ds: CfnDataSource;
+  /**
+   * creates a new resolver for this datasource and API using the given properties
+   */
+  createResolver(props: BaseResolverProps): Resolver;
+}
+
+/**
+ * Abstract AppSync datasource implementation.
+ *
+ * Do not use directly but use subclasses for concrete datasources
+ */
+export abstract class BaseDataSource extends Construct implements IDataSource {
```

Comment:
naming - should suffix with `Base` - the second comment re: not using it directly can be dropped if this is not exported.

Check out our [example resource](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/example-construct-library/lib/example-resource.ts#L151) for inspiration